docs/http/http-metrics.md

```diff
@@ -222,18 +239,24 @@ This metric is optional.
 | Attribute  | Type | Description  | Examples  | Requirement Level |
 |---|---|---|---|---|
 | `http.route` | string | The matched route (path template in the format used by the respective server framework). See note below [1] | `/users/:userID?`; `{controller}/{action}/{id?}` | Conditionally Required: If and only if it's available |
-| `http.request.method` | string | HTTP request method. [2] | `GET`; `POST`; `HEAD` | Required |
+| `http.request.error.reason` | string | Request failure reason. [2] | `timeout`; `name_resolution_error`; `server_certificate_invalid` | Conditionally Required: [3] |
+| `http.request.method` | string | HTTP request method. [4] | `GET`; `POST`; `HEAD` | Required |
 | `http.response.status_code` | int | [HTTP response status code](https://tools.ietf.org/html/rfc7231#section-6). | `200` | Conditionally Required: If and only if one was received/sent. |
 | [`network.protocol.name`](../general/attributes.md) | string | [OSI Application Layer](https://osi-model.com/application-layer/) or non-OSI equivalent. The value SHOULD be normalized to lowercase. | `amqp`; `http`; `mqtt` | Recommended |
-| [`network.protocol.version`](../general/attributes.md) | string | Version of the application layer protocol used. See note below. [3] | `3.1.1` | Recommended |
-| [`server.address`](../general/attributes.md) | string | Name of the local HTTP server that received the request. [4] | `example.com` | Opt-In |
-| [`server.port`](../general/attributes.md) | int | Port of the local HTTP server that received the request. [5] | `80`; `8080`; `443` | Opt-In |
+| [`network.protocol.version`](../general/attributes.md) | string | Version of the application layer protocol used. See note below. [5] | `3.1.1` | Recommended |
+| [`server.address`](../general/attributes.md) | string | Name of the local HTTP server that received the request. [6] | `example.com` | Opt-In |
+| [`server.port`](../general/attributes.md) | int | Port of the local HTTP server that received the request. [7] | `80`; `8080`; `443` | Opt-In |
 | [`url.scheme`](../url/url.md) | string | The [URI scheme](https://www.rfc-editor.org/rfc/rfc3986#section-3.1) component identifying the used protocol. | `http`; `https` | Required |
 
 **[1]:** MUST NOT be populated when this is not supported by the HTTP server framework as the route attribute should have low-cardinality and the URI path can NOT substitute it.
 SHOULD include the [application root](/docs/http/http-spans.md#http-server-definitions) if there is one.
 
-**[2]:** HTTP request method value SHOULD be "known" to the instrumentation.
+**[2]:** HTTP instrumentations that decide to report additional error codes SHOULD document them. Additional values MUST be a closed set of error reasons known at the development time of the instrumentation.
+MUST NOT be set if `http.response.status_code is set.
```

Comment:
Is this because we expect all providers to report errors only until headers read? What if a provider (other than .NET) chooses to extend the HTTP Span till the last byte of response, and something like an HTTP/2 protocol error occurs while reading the response body?

Reply:
Good point! Essentially we can have both, e.g. `http.response.status_code: 200`  and `http.request.error.reason: connection_reset` when response was not fully received (if the measurement includes response reading)